spyder/plugins/ipythonconsole/confpage.py

```diff
@@ -26,99 +26,148 @@ def setup_page(self):
-        comp_box = self.create_combobox(_("Completion:")+"   ", completers,
-                                        'completion_type')
+        comp_box = self.create_combobox(
+            _("Display:"), completers, 'completion_type')
+        jedi_box = newcb(
+            _("Use Jedi completion"),
+            "jedi_completer",
+            tip=_(
+                "Enable Jedi-based tab-completion in the IPython console;\n"
+                "similar to the greedy completer, "
+                "but without evaluating the code.\n"
+                "Allows completion of nested lists and similar.\n"
+                "Warning: "
+                "Slows down the console when working with large dataframes"
+            ),
+        )
+        greedy_box = newcb(
+            _("Use greedy completion"),
+            "greedy_completer",
+            tip=_(
+                "Enable <tt>Tab</tt> completion on elements of lists, "
+                "results of function calls and similar "
+                "<i>without</i> assigning them to a variable, "
+                "like <tt>li[0].&lt;Tab&gt;</tt> or "
+                "<tt>ins.meth().&lt;Tab&gt;</tt><br>"
+                "<b>Warning</b>: This can be unsafe because your code "
+                "is actually executed when you press <tt>Tab</tt>."
+            ),
+        )
 
         comp_layout = QVBoxLayout()
-        comp_layout.addWidget(comp_label)
         comp_layout.addWidget(comp_box)
+        comp_layout.addWidget(jedi_box)
+        comp_layout.addWidget(greedy_box)
         comp_group.setLayout(comp_layout)
 
-        # Source Code Group
-        source_code_group = QGroupBox(_("Source code"))
+        # Output group
+        output_group = QGroupBox(_("Output"))
 
         # Note: The maximum here is set to a relatively small value because
         # larger ones make Spyder sluggish.
         # Fixes spyder-ide/spyder#19091
         buffer_spin = self.create_spinbox(
-                _("Buffer:  "), _(" lines"),
-                'buffer_size', min_=-1, max_=5000, step=100,
-                tip=_("Set the maximum number of lines of text shown in the\n"
-                      "console before truncation. Specifying -1 disables it\n"
-                      "(not recommended!)"))
-
-        source_code_layout = QVBoxLayout()
-        source_code_layout.addWidget(buffer_spin)
-        source_code_group.setLayout(source_code_layout)
-
-        # --- Graphics ---
-        # Pylab Group
-        pylab_group = QGroupBox(_("Support for graphics (Matplotlib)"))
-        pylab_box = newcb(_("Activate support"), 'pylab')
-        autoload_pylab_box = newcb(
-            _("Automatically load Matplotlib and NumPy modules"),
+            _("Buffer:"),
+            _(" lines"),
+            'buffer_size',
+            min_=-1,
+            max_=5000,
+            step=100,
+            tip=_(
+                "The maximum number of scrollback lines shown in the Console."
```

Comment:
```suggestion
                "The maximum number of lines shown in the console at any time."
```

- `scrollback` is not an easy term to translate to other languages.
- I don't think we need to capitalize `console` because there can be many of them.

Reply:
> - `scrollback` is not an easy term to translate to other languages.

Its unfortunate, since it is the standard term for this in English. "Number of lines shown at any time" sounds like the number of lines shown simultaneously, which is not what this is.

> - I don't think we need to capitalize `console` because there can be many of them.

In that case, it should a "each console", not "the console" which would imply there is only one

```suggestion
                "The maximum number of output lines "
                "retained in each console at a time."
```